[INJICERT-1123] Add missing description for fields in API documentation

jackjain opened the pull request:
- Added description for fields in API documentation
- Added summary and description for the endpoints
- Added deprecated tag to /system-info/uploadCertificate endpoint

Stoplight link - changes are published for v0.12.0 : [https://mosip.stoplight.io/docs/inji-certify/branches/0.12.0]()

vishwa-vyom commented:
Remaining APIs comment:

system-info-controller:
1. why is upload signature certificate marked deprecated ?
2. upload signature request body, requesttime is with not following camelcase (Critical)

vc-issuance-controller:
1. /v1/certify/issuance/credential 
- do we support cwt ? If no, is there a JIRA for future ?
- do we support credentialSubject based filtering ? If no, is there a JIRA for future ?
- claims/doctype - usage related to credential format can be added to description 
- acceptance_token - do we support this now ?  If no, is there a JIRA for future ?
2. /credentials/.well-known/openid-credential-issuer vs /issuance/.well-known/openid-credential-issuer - why do we have 2 endpoints for metadata and which one is latest ? (Critical)
3. For all the wellknowns should like .well-known/openid-credential-issuer and /.well-known/did.json, should we move it to a new controller called well-known-controller both in openapi doc and code ? (Critical)

Ledger search:
1. Any particular reason for not following the controller convention ? Lets try to maintain consistency

Credential status:
1. /v1/certify/issuance/credential/status change to /v1/certify/credentials/status - to be compliant with W3C VC API spec. (Critical)

Status List:
1. Rename /v1/certify/status-list/{id} to /v1/certify/credentials/status-list/{id} (Critical)
2. Move status list under the Credential status section
3. Do we support timestamp query parameter ?
4. Do we support all the headers  in the code ?
5. Body description says StatusList2021Credential which is not correct
6. Response body content type is application/json ?

jackjain commented:
Review comment ref : https://github.com/mosip/inji-certify/pull/375#issuecomment-3086626362

system-info-controller:
1. why is upload signature certificate marked deprecated ? _**- Removed**_
2. upload signature request body, requesttime is with not following camelcase (Critical) - _**Need a JIRA card to track this, Varaniya will share**_

vc-issuance-controller:
1. /v1/certify/issuance/credential
* do we support cwt ? If no, is there a JIRA for future ? - _**https://mosip.atlassian.net/browse/MOSIP-42630  - Assigned to Keymanager currently**_
* do we support credentialSubject based filtering ? If no, is there a JIRA for future ? - _**Varaniya to give a JIRA card**_
* claims/doctype - usage related to credential format can be added to description  - **_Added reference of credential format in the claims and docType description_**
* acceptance_token - do we support this now ? If no, is there a JIRA for future ? - **_We don’t support this at this point, need a JIRA to track this._**
1. /credentials/.well-known/openid-credential-issuer vs /issuance/.well-known/openid-credential-issuer - why do we have 2 endpoints for metadata and which one is latest ? (Critical) - _**removed /credentials/.well-known/openid-credential-issuer, /issuance/.well-known/openid-credential-issuer  is now deprecatred and now internally pointing to credential-configuration service.**_
2. For all the wellknowns should like .well-known/openid-credential-issuer and /.well-known/did.json, should we move it to a new controller called well-known-controller both in openapi doc and code ? (Critical) - **_Created a new well-known controller_** 
Ledger search:
1. Any particular reason for not following the controller convention ? Lets try to maintain consistency - **_Created a separate controller for ledger search_**
Credential status:
1. /v1/certify/issuance/credential/status change to /v1/certify/credentials/status - to be compliant with W3C VC API spec. (Critical) - _**Created the credentialStatus controller**_

Status List:
1. Rename /v1/certify/status-list/{id} to /v1/certify/credentials/status-list/{id} (Critical) - _**Moved it to CredentialStatus controller as part of critical code review changes**_
2. Move status list under the Credential status section - **_Moved_**
3. Do we support timestamp query parameter ? - **_Removed the unsupported query param_**
4. Do we support all the headers in the code ? - **_No, removed from documentation_**
5. Body description says StatusList2021Credential which is not correct - **_Done, replaced it with BitstringStatusListCredential_**
6. Response body content type is application/json ? - **_yes_**

Review comment ref : https://github.com/mosip/inji-certify/pull/375#issuecomment-3178113268

1. Try hitting endpoints from swagger - see if the response comes back properly  - **_we are able to hit URLs from swagger/stoplight, the ones which require authentication have copied the request and validated them via postman_**
2. signatureAlgo algo enums not coming up in swagger, also add possible combinations of the signatureAlgo - signatureCryptoSuite - **_updated the type to string to fix the issue_**
3. Add VCIssuance example for each type in credential config - **_Added example ldp_vc and mso_mdoc, not supported for vc+sd-jwt currently**
4. displayOrder - indicate the link between it and credentialSubjectDefinition - **_added the reference in  displayOrder field_**
5. Remove deprecated tag from upload signature certificate - **_removed the tag_** 


Pending JIRAs - Will add them here once they shared by Varaniya: 
upload signature request body, requesttime is with not following camelcase (Critical) 
do we support credentialSubject based filtering ? If no, is there a JIRA for future ? 
acceptance_token - do we support this now ? If no, is there a JIRA for future ?

vishwa-vyom commented:
@jackjain 
I can see multiple line items that says needs JIRA creation, not sure how we are making sure that will not loose track of these findings post release of this version. cc: @Varaniya @preetighongal 

"do we support cwt ? If no, is there a JIRA for future ? - https://mosip.atlassian.net/browse/MOSIP-42630 - Assigned to Keymanager currently" - this jira points to only mdoc, but cwt based proof type generation part I am not able to see in this JIRA

"Response body content type is application/json ? - yes" - I think we should have a jira on this also. Reference - https://www.w3.org/TR/vc-bitstring-status-list/#media-types



> Review comment ref : [#375 (comment)](https://github.com/mosip/inji-certify/pull/375#issuecomment-3086626362)
> 
> system-info-controller:
> 
> 1. why is upload signature certificate marked deprecated ? _**- Removed**_
> 2. upload signature request body, requesttime is with not following camelcase (Critical) - _**Need a JIRA card to track this, Varaniya will share**_
> 
> vc-issuance-controller:
> 
> 1. /v1/certify/issuance/credential
> 
> * do we support cwt ? If no, is there a JIRA for future ? - _**https://mosip.atlassian.net/browse/MOSIP-42630  - Assigned to Keymanager currently**_
> * do we support credentialSubject based filtering ? If no, is there a JIRA for future ? - _**Varaniya to give a JIRA card**_
> * claims/doctype - usage related to credential format can be added to description  - **_Added reference of credential format in the claims and docType description_**
> * acceptance_token - do we support this now ? If no, is there a JIRA for future ? - **_We don’t support this at this point, need a JIRA to track this._**
> 
> 1. /credentials/.well-known/openid-credential-issuer vs /issuance/.well-known/openid-credential-issuer - why do we have 2 endpoints for metadata and which one is latest ? (Critical) - _**removed /credentials/.well-known/openid-credential-issuer, /issuance/.well-known/openid-credential-issuer  is now deprecatred and now internally pointing to credential-configuration service.**_
> 2. For all the wellknowns should like .well-known/openid-credential-issuer and /.well-known/did.json, should we move it to a new controller called well-known-controller both in openapi doc and code ? (Critical) - **_Created a new well-known controller_**
>    Ledger search:
> 3. Any particular reason for not following the controller convention ? Lets try to maintain consistency - **_Created a separate controller for ledger search_**
>    Credential status:
> 4. /v1/certify/issuance/credential/status change to /v1/certify/credentials/status - to be compliant with W3C VC API spec. (Critical) - _**Created the credentialStatus controller**_
> 
> Status List:
> 
> 1. Rename /v1/certify/status-list/{id} to /v1/certify/credentials/status-list/{id} (Critical) - _**Moved it to CredentialStatus controller as part of critical code review changes**_
> 2. Move status list under the Credential status section - **_Moved_**
> 3. Do we support timestamp query parameter ? - **_Removed the unsupported query param_**
> 4. Do we support all the headers in the code ? - **_No, removed from documentation_**
> 5. Body description says StatusList2021Credential which is not correct - **_Done, replaced it with BitstringStatusListCredential_**
> 6. Response body content type is application/json ? - **_yes_**
> 
> Review comment ref : [#375 (comment)](https://github.com/mosip/inji-certify/pull/375#issuecomment-3178113268)
> 
> 1. Try hitting endpoints from swagger - see if the response comes back properly  - **_we are able to hit URLs from swagger/stoplight, the ones which require authentication have copied the request and validated them via postman_**
> 2. signatureAlgo algo enums not coming up in swagger, also add possible combinations of the signatureAlgo - signatureCryptoSuite - **_updated the type to string to fix the issue_**
> 3. Add VCIssuance example for each type in credential config - **_Added example ldp_vc and mso_mdoc, not supported for vc+sd-jwt currently**
> 4. displayOrder - indicate the link between it and credentialSubjectDefinition - **_added the reference in  displayOrder field_**
> 5. Remove deprecated tag from upload signature certificate - **_removed the tag_**
> 
> Pending JIRAs - Will add them here once they shared by Varaniya: upload signature request body, requesttime is with not following camelcase (Critical) do we support credentialSubject based filtering ? If no, is there a JIRA for future ? acceptance_token - do we support this now ? If no, is there a JIRA for future ?

Varaniya201 commented:
@vishwa-vyom I have checked on the same with Taheer, as part of COSE signing, he will be including cwt as well in the same story, this was reviewed by Taheer
